### ISSUE TITLE
After upgrading to 24.13.2, Windows NSIS installer Failed to uninstall old application files when there is an app running

### ISSUE THREAD

**Issue by zaichaopan-alianza** [electron-userland/electron-builder#8131]
* **Electron-Builder Version**: 24.13.2

* **Node Version**: 18.14.0

* **Electron Version**: 29.1.1
* **Electron Type (current, beta, nightly)**: current

* **Target**: Windows

After upgrading to 24.13.2, when installing nsis installer when app is running, it shows below error "**xxxx cannot be closed. Please close it manually and click Retry to continue**"

![electron-quick-start-is-running](https://github.com/electron-userland/electron-builder/assets/66754912/c0c013ee-6fec-4bed-8baf-e44df23b69c7)

Click **Retry** will keep showing the dialog again. When click **Cancel**, it shows "**old application files cannot be uninstalled**"
![electron-quick-start-is-cannot-uninstall-old-app](https://github.com/electron-userland/electron-builder/assets/66754912/310c2f53-2fd7-493d-920f-1a66bba6ac8f)

**Versions before 24.13.2 do not have this issue**

When try previous version **24.13.1**,  It shows dialog **xxx is running, click OK to close**. After click **OK**, running app will close and install will continue.

![electron-quick-start-13 1](https://github.com/electron-userland/electron-builder/assets/66754912/fbd81194-f143-44ec-823e-3b9991c7481a)

### Step to reproduce
- upgrade electron-builder to 24.13.2 and create a NSIS installer
- install the installer and have the app running
- click the installer again to install or make another new installer to install

Seems like the fix in https://github.com/electron-userland/electron-builder/pull/8059 introduce this issue.  Have anyone run into the same issue?

Thx a million !

**Comment by mmaietta** [comment]
What's your `nsis` electron-builder config? I'll try and repro this locally

**Comment by zaichaopan-alianza** [comment]
thank you so much @mmaietta I have the repo https://github.com/zaichaopan-alianza/electron-builder-24.13.2-nsis-installer you can try. It is from [electron-quick-start](https://github.com/electron/electron-quick-start). My config is very simple. It just 

```js
{
  "name": "electron-quick-start",
  "version": "1.0.0",
  "description": "A minimal Electron application",
  "main": "main.js",
  "build": {
    "appId": "com.example.electron-quick-start"
  },
  "scripts": {
    "start": "electron .",
    "pack": "electron-builder --dir",
    "dist": "electron-builder"
  },
  "repository": "https://github.com/electron/electron-quick-start",
  "keywords": [
    "Electron",
    "quick",
    "start",
    "tutorial",
    "demo"
  ],
  "author": "GitHub",
  "license": "CC0-1.0",
  "devDependencies": {
    "electron": "^29.1.1",
    "electron-builder": "24.13.2"
  }
}
````

**Comment by mmaietta** [comment]
Thanks for the repo! Really appreciate it.

Will take a look this week when I find some solid time outside of work

**Comment by beyondkmp** [comment]
SYSTEMROOT = "c:\windows", The path c:\windows contains special characters, so it needs to be enclosed in double quotes to work properly.

```
!define SYSTEMROOT "$%SYSTEMROOT%"
nsExec::Exec '"${SYSTEMROOT}\System32\cmd.exe" /c tasklist /FI "USERNAME eq %USERNAME%" /FI "IMAGENAME eq ${_FILE}" /FO csv | "${SYSTEMROOT}\System32\find.exe" "${_FILE}"'
```

**Comment by zaichaopan-alianza** [comment]
@beyondkmp I tried the fix https://github.com/electron-userland/electron-builder/pull/8133. The issue is no longer reproduced. Thanks a lot!

**Comment by mmaietta** [comment]
Releasing in `25.0.0-alpha.4`

**Comment by sumeet-singh04** [comment]
Would this fix be backported to 24.13.3 ?

**Comment by mmaietta** [comment]
Unfortunately, AFAIK it's not possible to backport fixes to previous versions due to the @changesets CI/CD approach. It's only able to roll forward AFAICT. Happy to be disproved otherwise though!

**Comment by qiaoqiaoweb** [comment]
Upgrading to 25.0.0 alpha. 4 still has this issue as there is no custom NISI configuration

**Comment by indutny-signal** [comment]
Was the fix for this released already?

**Comment by mmaietta** [comment]
@indutny-signal The fix that was implemented was released in `25.0.0-alpha.4`

**Comment by indutny-signal** [comment]
I was hoping for a stable release, but currently fighting my way through upgrading to 25.0.0 😂 Universal builds appear to be a bit hosed.

**Comment by mmaietta** [comment]
@indutny-signal yeah, I plan on converting it to stable soon, I was just hoping for more adoption than the ~7% it currently is 😅

Can you open a new GH issue for the universal build issue you mention? That particularly confuses me.

**Comment by finnlaymfarmor** [comment]
Hi @mmaietta ,

We're currently running electron-builder@25.0.5, and getting the same error. I've tried a number of different fixes across GH Issues, SO, and asking LLM's but nothing seems to be remedying. Any ideas?

**Comment by beyondkmp** [comment]
@finnlaymfarmor @1007912932  Can you provide a minimal reproducible demo code?

**Comment by 1007912932** [comment]
electron-builder哪个版本解决了吗，我看8133关闭了

**Comment by 1007912932** [comment]
能加个微信沟通下吗？
我的微信号：luffy_is_fool

**Comment by LWD8** [comment]
降回22.14.13就可以了，升级electron-builder@25也不行

**Comment by erickaito153** [comment]
I am using electron-builder version 25.0.5 and got the same issue. I have a function to update my application domestically downloading the exe file from the server when a new version exists which will then be saved to the same installed directory of the application with a different name. Then I click the install button on the app, it will invoke the app.quit() from electron. Finally, the installation replacement will start and the new version will be installed. However, when clicking the install button, I got the issue as the topic. Do you guys have any solutions for this?

**Comment by cgarrovillo** [comment]
This issue needs to be reopened as it seems we have not come into a common conclusion that this is fixed @mmaietta

**Comment by brandonin** [comment]
Received this issue utilizing:
    "electron": "^36.3.2",
    "electron-builder": "^26.0.12",

**Comment by mmaietta** [comment]
Could anyone please provide a minimum repro repo of this issue on the latest `26.0.x` release of electron-builder? Something I can debug further with?

I think this might actually be resolved through this PR https://github.com/electron-userland/electron-builder/pull/9069 but it requires Powershell to be present/enabled in order for the new/separate logic flow to take effect.

**Comment by iamobj** [comment]
> Could anyone please provide a minimum repro repo of this issue on the latest `26.0.x` release of electron-builder? Something I can debug further with?
> 
> I think this might actually be resolved through this PR [#9069](https://github.com/electron-userland/electron-builder/pull/9069) but it requires Powershell to be present/enabled in order for the new/separate logic flow to take effect.

"electron": "^35.5.1",
"electron-builder": "^26.0.12"

This configuration I use has this problem, but I personally think it has nothing to do with the configuration, as if it has something to do with the computer environment, because some computers will not appear, some will

```
win: {
    icon: 'resources/icon.ico',
    requestedExecutionLevel: 'asInvoker',
    target: [
      {
        target: 'nsis',
      },
    ],
    artifactName: '${productName}-${version}-${os}-${arch}-setup.${ext}',
  },
  nsis: {
    oneClick: false,
    allowToChangeInstallationDirectory: true,
    installerIcon: 'resources/icon.ico',
    uninstallerIcon: 'resources/icon.ico',
    createDesktopShortcut: true,
    createStartMenuShortcut: true,
    include: 'build/installer.nsh',
    shortcutName: '${productName}',
    uninstallDisplayName: '${productName}',
  }
```

**Comment by iamobj** [comment]
There is more discussion on this issue here https://github.com/electron-userland/electron-builder/issues/6865

**Comment by GabrielNSD** [comment]
Hi there! Could this be related to the Windows version?
I hadn’t encountered this issue before, but after a Windows update, it started happening. So far, I’ve noticed this issue occurring on Windows 11 Pro, Version 24H2 (Build 26100.4351).

**Comment by zulqarnain-sabir** [comment]
still facing this issue, any recommendations?
